python/kserve/kserve/model.py

```diff
@@ -218,23 +222,36 @@ async def preprocess(self, payload: Union[Dict, CloudEvent, ModelInferRequest],
 
         return response
 
-    def postprocess(self, response: Union[Dict, ModelInferResponse], headers: Dict[str, str] = None) -> Dict:
-        """The postprocess handler can be overridden for inference response transformation
+    def postprocess(self, response: Union[Dict, InferResponse, ModelInferResponse], headers: Dict[str, str] = None) \
+            -> Union[Dict, ModelInferResponse]:
+        """The postprocess handler can be overridden for inference response transformation.
+        The default implementation converts the v2 infer response types to gRPC or REST.
+        For gRPC request it converts InferResponse to gRPC message or directly returns ModelInferResponse from
```

Comment:
do you mean to have InferRequest here instead of ModelInferRequest since the name change?

Reply:
oh wait I see there are two types now.. I see.. Im thinking the naming is a little confusing but right now I can't think of something better so I will keep thinking